packages/browser/__tests__/Session.spec.ts

```diff
@@ -0,0 +1,205 @@
+      expect(mySession.info.isLoggedIn).toEqual(false);
+      expect(mySession.info.sessionId).toBeDefined();
+    });
+
+    it("does not generate a session id if one is provided", () => {
+      const mySession = new Session({}, "mySession");
+      expect(mySession.info.sessionId).toEqual("mySession");
+    });
+
+    it("accepts input storage", async () => {
+      const insecureStorage = mockStorage({});
+      const secureStorage = mockStorage({});
+      const mySession = new Session({
+        insecureStorage,
+        secureStorage,
+      });
+      const clearSecureStorage = jest.spyOn(secureStorage, "delete");
+      const clearInsecureStorage = jest.spyOn(insecureStorage, "delete");
+      await mySession.logout();
+      expect(clearSecureStorage).toHaveBeenCalled();
+      expect(clearInsecureStorage).toHaveBeenCalled();
+    });
+
+    it("accepts session info", () => {
+      const mySession = new Session({
+        sessionInfo: {
+          sessionId: "mySession",
+          isLoggedIn: false,
+          webId: "https://some.webid",
+        },
+      });
+      expect(mySession.info.isLoggedIn).toEqual(false);
+      expect(mySession.info.sessionId).toEqual("mySession");
+      expect(mySession.info.webId).toEqual("https://some.webid");
+    });
+  });
+
+  describe("login", () => {
+    it("wraps up ClientAuthentication login", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnLogin = jest.spyOn(clientAuthentication, "login");
+      const mySession = new Session({ clientAuthentication });
+      await mySession.login({});
+      expect(clientAuthnLogin).toHaveBeenCalled();
+    });
+  });
+
+  describe("logout", () => {
+    it("wraps up ClientAuthentication logout", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnLogout = jest.spyOn(clientAuthentication, "logout");
+      const mySession = new Session({ clientAuthentication });
+      await mySession.logout();
+      expect(clientAuthnLogout).toHaveBeenCalled();
+    });
+  });
+
+  describe("fetch", () => {
+    it("wraps up ClientAuthentication fetch", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnFetch = jest.spyOn(clientAuthentication, "fetch");
+      const mySession = new Session({ clientAuthentication });
+      await mySession.fetch("https://some.url");
+      expect(clientAuthnFetch).toHaveBeenCalled();
+    });
+  });
+
+  describe("handleincomingRedirect", () => {
+    it("wraps up ClientAuthentication handleIncomingRedirect", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnHandle = jest.spyOn(
+        clientAuthentication,
+        "handleIncomingRedirect"
+      );
+      const mySession = new Session({ clientAuthentication });
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(clientAuthnHandle).toHaveBeenCalled();
```

Comment:
Ditto - i.e. is there specific behaviour (from our mocked `clientAuthentication`) that we can assert on in the Session instance here instead of expecting a specific method to have been called?